### ISSUE TITLE
ERROR: (gcloud.run.deploy) Invalid value for property [api_endpoint_overrides/run]

### ISSUE THREAD

**Issue by rodt7** [google-github-actions/deploy-cloudrun#297]
### TL;DR

Hi team, thanks for sharing the wonderful solution there.

I got an error complaining about the `Invalid value for property [api_endpoint_overrides/run]`

### Observed behavior

```
Error: failed to execute gcloud command `gcloud run deploy automation-dummy-api --image $GAR_LOCATION-docker.pkg.dev/$PROJECT_ID/$REPOSITORY/$IMAGE:$GITHUB_SHA --quiet --platform managed --region $GAR_LOCATION --project XXXX --format json`: ERROR: (gcloud.run.deploy) Invalid value for property [api_endpoint_overrides/run]: The endpoint_overrides property must be an absolute URI beginning with http:// or https:// and ending with a trailing '/'. [https://$GAR_LOCATION-run.googleapis.com/] is not a valid endpoint override.

```

### Action YAML

```yaml
name: Deployment

on:
  workflow_dispatch:
    inputs:
        version:
          description: 'The version to deploy'
          required: true

env:
  PROJECT_ID: XXXX # ${{ secrets.GCP_PROJECT }}
  GAR_LOCATION: australia-southeast1 # TODO: update region of the Artifact Registry
  REPOSITORY: automation # TODO: update to Artifact Registry docker repository
  IMAGE: automation-template

permissions:
    contents: 'read'
    id-token: 'write'

jobs:
  deployment:
    name: Deployment
    runs-on: ubuntu-latest
    environment: production

    steps:
      - uses: actions/checkout@v2
      - id: auth
        name: 'Authenticate to Google Cloud'
        uses: 'google-github-actions/auth@v0'
        with:
          token_format: 'access_token'
          workload_identity_provider: 'projects/741041240XXX/locations/global/workloadIdentityPools/github-action-pool3/providers/github-action-provider3'
          service_account: 'github-action-sa3@XXXX.iam.gserviceaccount.com'

      - name: Docker configuration
        run: |-
          echo ${{steps.auth.outputs.access_token}} | docker login -u oauth2accesstoken --password-stdin https://$GAR_LOCATION-docker.pkg.dev

      - name: Build
        run: |-
          docker build \
            --tag "$GAR_LOCATION-docker.pkg.dev/$PROJECT_ID/$REPOSITORY/$IMAGE:$GITHUB_SHA" \
            --build-arg GITHUB_SHA="$GITHUB_SHA" \
            --build-arg GITHUB_REF="$GITHUB_REF" \
            .

      - name: Run Tests
        run: |-
          docker run "$GAR_LOCATION-docker.pkg.dev/$PROJECT_ID/$REPOSITORY/$IMAGE:$GITHUB_SHA" pytest tests

      # Push the Docker image to Google Artifact Registry
      - name: Publish
        run: |-
          docker push "$GAR_LOCATION-docker.pkg.dev/$PROJECT_ID/$REPOSITORY/$IMAGE:$GITHUB_SHA"

      - name: 'Deploy to Cloud Run'
        # You may pin to the exact commit or the version.
        # uses: google-github-actions/deploy-cloudrun@ff8b38330cf81843db9cff366e335c34aa467c44
        uses: google-github-actions/deploy-cloudrun@main
        with:
          # Name of the container image to deploy (e.g. gcr.io/cloudrun/hello:latest).
          # Required if not using a service YAML.
          image: $GAR_LOCATION-docker.pkg.dev/$PROJECT_ID/$REPOSITORY/$IMAGE:$GITHUB_SHA
          # ID of the service or fully qualified identifier for the service.
          # Required if not using a service YAML.
          service: automation-dummy-api
          # Region in which the resource can be found.
          region: $GAR_LOCATION
          # List of key-value pairs to set as environment variables in the format:
          # KEY1=VALUE1,KEY2=VALUE2. All existing environment variables will be
          # removed first.
          # env_vars: # optional
          # YAML serivce description for the Cloud Run service.
          # metadata: # optional
          # The GCP project ID. Overrides project ID set by credentials.
          # project_id: # optional
```
```


### Log output

```text
Error: failed to execute gcloud command `gcloud run deploy automation-dummy-api --image $GAR_LOCATION-docker.pkg.dev/$PROJECT_ID/$REPOSITORY/$IMAGE:$GITHUB_SHA --quiet --platform managed --region $GAR_LOCATION --project XXXX --format json`: ERROR: (gcloud.run.deploy) Invalid value for property [api_endpoint_overrides/run]: The endpoint_overrides property must be an absolute URI beginning with http:// or https:// and ending with a trailing '/'. [https://$GAR_LOCATION-run.googleapis.com/] is not a valid endpoint override.

```
```


### Additional information

_No response_

**Comment by rodt7** [comment]
![image](https://user-images.githubusercontent.com/98290132/151911255-60aa8785-c5a7-4047-868a-53983ff4c7ba.png)

**Comment by verbanicm** [comment]
@rodt7 I believe you need to use the [expression](https://docs.github.com/en/actions/learn-github-actions/expressions) syntax for those environment variables. For a list of contexts available you can see the docs [here](https://docs.github.com/en/actions/learn-github-actions/environment-variables).

Here is a quick example for your use case:

```yaml
- name: "Deploy to Cloud Run"
  # You may pin to the exact commit or the version.
  # uses: google-github-actions/deploy-cloudrun@ff8b38330cf81843db9cff366e335c34aa467c44
  uses: google-github-actions/deploy-cloudrun@main
  with:
    # Name of the container image to deploy (e.g. gcr.io/cloudrun/hello:latest).
    # Required if not using a service YAML.
    image: '${{ env.GAR_LOCATION }}-docker.pkg.dev/${{ env.PROJECT_ID }}/${{ env.REPOSITORY }}/${{ env.IMAGE }}:${{ env.GITHUB_SHA }}'
    # ID of the service or fully qualified identifier for the service.
    # Required if not using a service YAML.
    service: automation-dummy-api
    # Region in which the resource can be found.
    region: '${{ env.GAR_LOCATION }}'
```

**Comment by rodt7** [comment]
I see. Thank you!
